src/main/java/com/dataguise/saas/service/OverviewService.java

```diff
@@ -0,0 +1,316 @@
+                throw new InternalServerException(ErrorConstants.ERROR_PARSING_SOURCE_LIST.getErrorMessage());
+            }
+        }
+        return sourceHierarchyExplorerServiceService.getFormattedOverviewSourceList(sourcesDTOList);
+    }
+
+    /**
+     * Get dbms cloud databases of a selected source from dashboard api using ConnectionHierarchyExplorer service.
+     *
+     * @return a list of databases for a source in formatted structure.
+     */
+    public List<SourceTreeViewUIFormatterDTO> getDashboardDatabases(Integer connectionId) throws BadRequestException, BadGatewayException {
+        List<SourceTreeViewUIFormatterDTO> formattedDatabaseList = null;
+        if (connectionId == null) {
+            throw new BadRequestException("Connection " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage());
+        }
+        try {
+            List<DrillDownDetailedResponse> detailedResponseList = restDriver.getDatabasesForConnection(connectionId);
+            formattedDatabaseList = sourceHierarchyExplorerServiceService.getFormattedOverviewDatabaseList(detailedResponseList, connectionId);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        return formattedDatabaseList;
+    }
+
+
+    /**
+     * Get all sources from controller using ConnectionHierarchyExplorer service.
+     *
+     * @return all source list response in formatted structure .
+     */
+    public List<SourceTreeViewUIFormatterDTO> getSources() throws BadGatewayException, InternalServerException {
+        List<SourcesDTO> sourcesDTOList = new ArrayList<>();
+        List<ConnectionDTO> dgConnectionList = connectionService.getConnections(Optional.empty());
+        //Only detection sources
+        dgConnectionList = dgConnectionList.stream().
+                filter(dgConnection -> dgConnection.getConnection().getConnectionAgent().equalsIgnoreCase("D") || dgConnection.getConnection().getConnectionAgent().equalsIgnoreCase("M"))
+                .collect(Collectors.toList());
+
+        for (ConnectionDTO conn : dgConnectionList) {
+            DgConnection dgConnection = conn.getConnection();
+            SourcesDTO sourceDTO = new SourcesDTO();
+            sourceDTO.setName(dgConnection.getConnectionName());
+            sourceDTO.setSourceName(dgConnection.getConnectionName());
+            sourceDTO.setSourceId(dgConnection.getConnectionId());
+            sourceDTO.setServerType(dgConnection.getLocation());
+            sourceDTO.setAgentType(dgConnection.getConnectionAgent());
+            sourcesDTOList.add(sourceDTO);
+        }
+        List<SourceTreeViewUIFormatterDTO> ControllerSourceTreeViewUIFormatterDTOList =
+                sourceHierarchyExplorerServiceService.getFormattedOverviewSourceList(sourcesDTOList);
+
+        List<SourceTreeViewUIFormatterDTO> DashboardSourceTreeViewUIFormatterDTOList = getDashboardSources();
```

Comment:
**Reporter**: PMD    
    **Rule**: LocalVariableNamingConventions
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/OverviewService.java L176
    
    The local variable name &#39;DashboardSourceTreeViewUIFormatterDTOList&#39; doesn&#39;t match &#39;[a-z][a-zA-Z0-9]*&#39;&#10;&#10;&#10;Code Style https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#localvariablenamingconventions
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-222020645>*